Simplify Release Process

asonnenschein opened the pull request:
This PR simplifies the SDK release process by reducing the number of release branches, naming conventions, and number of steps to release:
- All releases are now tagged from the `main` branch.  Developers should still create a release branch, but only to bump the version number and merge to `main`.  The previous release process required tagging releases from release branches, and making multiple updates to the release version number both before and after tagging the release.
- Included new workflow for releasing via `gh` CLI.
- Abandon release naming conventions.  Developers should follow vanilla semantic versioning when tagging releases.  The previous release process included naming convention prefixes/suffixes like `maint-1.0.0` and `1.0.0dev`.
- Remove internal Planet Slack announcement steps.  I am currently working with folks to integrate GitHub notifications from this repository with our internal company Slack, which will announce releases automatically as they are tagged.

This MR doesn't make any changes to automated CI workflows used in the release process; the changes proposed here should not have any side effects with existing CI workflows.

tbarsballe commented on RELEASE.md:
```diff
@@ -2,76 +2,33 @@
+2. Create a PR for the release branch and merge into `main`
+3. Create a new GitHub release:
+   * Set tag to release version
+   * Set target to `main`
```
You can also tag in git and push the tag
e.g. 
* `git tag <release_version>`
* `git push <release_version>`
* Create a new github release with the target as the tag

asonnenschein replied:
MR is updated with a workflow for tagging a new release via the `gh` CLI.  I tried tagging a release via `git tag` and while it does indeed create a new tag, it doesn't automatically create a new release entry.  With `gh release create` I can tag a new release and add release notes, which will create a new release entry in the releases UI in addition to the new tag.

tbarsballe reviewed (approved):
One minor comment - you can also tag from git directly and release from the tag

asonnenschein commented:
Made some updates, per feedback during office hours:
- Updated the release docs with a workflow for tagging a new release via the `gh` CLI
- Added `CHANGES.txt` back
- Updated `Versions and Stability` section in README with Ian's text, and updated the `Versions and Stability` section in RELEASES with a simple note about semantic versioning.
